src/Microsoft.Identity.Web/AppServicesAuth/AppServicesAuthenticationTokenAcquisition.cs

```diff
@@ -13,181 +13,181 @@
-            IEnumerable<string> scopes,
-            string? tenantId = null,
-            string? userFlow = null,
-            ClaimsPrincipal? user = null,
-            TokenAcquisitionOptions? tokenAcquisitionOptions = null)
-        {
-            string? idToken = AppServicesAuthenticationInformation.GetIdToken(CurrentHttpContext?.Request?.Headers!);
-            ClaimsPrincipal? userClaims = AppServicesAuthenticationInformation.GetUser(CurrentHttpContext?.Request?.Headers!);
-            string accessToken = await GetAccessTokenForUserAsync(scopes, tenantId, userFlow, user, tokenAcquisitionOptions).ConfigureAwait(false);
-            string expiration = userClaims.FindFirstValue("exp");
-            DateTimeOffset dateTimeOffset = (expiration != null)
-                ? DateTimeOffset.FromUnixTimeSeconds(long.Parse(expiration, CultureInfo.InvariantCulture))
-                : DateTimeOffset.Now;
-
-            AuthenticationResult authenticationResult = new AuthenticationResult(
-                accessToken,
-                isExtendedLifeTimeToken: false,
-                userClaims?.GetDisplayName(),
-                dateTimeOffset,
-                dateTimeOffset,
-                userClaims?.GetTenantId(),
-                userClaims != null ? new Account(userClaims) : null,
-                idToken,
-                scopes,
-                tokenAcquisitionOptions != null ? tokenAcquisitionOptions.CorrelationId : Guid.Empty);
-            return authenticationResult;
-        }
-
-        /// <inheritdoc/>
-        public async Task ReplyForbiddenWithWwwAuthenticateHeaderAsync(IEnumerable<string> scopes, MsalUiRequiredException msalServiceException, HttpResponse? httpResponse = null)
-        {
-            // Not implemented for the moment
-            throw new NotImplementedException();
-        }
-
-        /// <inheritdoc/>
-        public Task<AuthenticationResult> GetAuthenticationResultForAppAsync(string scope, string? tenant = null, TokenAcquisitionOptions? tokenAcquisitionOptions = null)
-        {
-            throw new NotImplementedException();
-        }
+		public async Task<AuthenticationResult> GetAuthenticationResultForUserAsync(
+			IEnumerable<string> scopes,
+			string? tenantId = null,
+			string? userFlow = null,
+			ClaimsPrincipal? user = null,
+			TokenAcquisitionOptions? tokenAcquisitionOptions = null)
+		{
+			string? idToken = AppServicesAuthenticationInformation.GetIdToken(CurrentHttpContext?.Request?.Headers!);
+			ClaimsPrincipal? userClaims = AppServicesAuthenticationInformation.GetUser(CurrentHttpContext?.Request?.Headers!);
+			string accessToken = await GetAccessTokenForUserAsync(scopes, tenantId, userFlow, user, tokenAcquisitionOptions).ConfigureAwait(false);
+			string expiration = userClaims.FindFirstValue("exp");
+			DateTimeOffset dateTimeOffset = (expiration != null)
+				? DateTimeOffset.FromUnixTimeSeconds(long.Parse(expiration, CultureInfo.InvariantCulture))
```

Comment:
long.parse could throw if the claim has a broken expiration string. long.TryParse would be safer.